src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,162 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
-    @Override
-    public void add(T value) {
+    private static final int ZERO_SIZE = 0;
+    private static final int HALF_SIZE = 2;
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T data;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(T data) {
+            this.data = data;
+        }
     }
```

Comment:
Don't use getters and setters in class Node. Since Node is an inner class, we can access its private fields directly from the outer class.